tests/utils/test_serializers.py

```diff
@@ -0,0 +1,102 @@
+import datetime
+from typing import Any, Dict, Type
+
+import pytest
+
+from reflex.utils import serializers
+
+
+@pytest.mark.parametrize(
+    "type_,expected",
+    [
+        (str, True),
+        (dict, True),
+        (Dict[int, int], True),
+    ],
+)
+def test_has_serializer(type_: Type, expected: bool):
+    """Test that has_serializer returns the correct value.
+
+
+    Args:
+        type_: The type to check.
+        expected: The expected result.
+    """
+    assert serializers.has_serializer(type_) == expected
+
+
+@pytest.mark.parametrize(
+    "type_,expected",
+    [
+        (str, serializers.serialize_str),
+        (dict, serializers.serialize_dict),
+        (Dict[int, int], serializers.serialize_dict),
+        (datetime.datetime, serializers.serialize_datetime),
+        (datetime.date, serializers.serialize_datetime),
+        (datetime.time, serializers.serialize_datetime),
+        (datetime.timedelta, serializers.serialize_datetime),
+    ],
+)
+def test_get_serializer(type_: Type, expected: serializers.Serializer):
+    """Test that get_serializer returns the correct value.
+
+
+    Args:
+        type_: The type to check.
+        expected: The expected result.
+    """
+    assert serializers.get_serializer(type_) == expected
+
+
+def test_add_serializer():
+    """Test that adding a serializer works."""
+
+    def serialize_test(value: int) -> str:
+        """Serialize an int to a string.
+
+        Args:
+            value: The value to serialize.
+
+        Returns:
+            The serialized value.
+        """
+        return str(value)
+
+    # Initially there should be no serializer for int.
+    assert not serializers.has_serializer(int)
+    assert serializers.serialize(5) is None
+
+    # Register the serializer.
+    serializers.serializer(serialize_test)
```

Comment:
```suggestion
    assert serializers.serializer(serialize_test) is serialize_test
```

just a bit of a regression test to make sure `serializer` isn't munging the object it decorates